### PR TITLE
Revert "Revert "DT-1108: 🔥 Remove username validation for client credentials""

### PR DESCRIPTION
**Pull request by petergphillips**
Reverts ministryofjustice/hmpps-auth#533

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 plugins {
   id("uk.gov.justice.hmpps.gradle-spring-boot") version "1.0.2"
+  id("groovy")
```
Comment (petergphillips) [inline build.gradle.kts:3]: add in missing groovy plugin for the tests

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/JWTTokenEnhancer.java [hunk 2]
```diff
@@ -95,29 +93,34 @@ public OAuth2AccessToken enhance(final OAuth2AccessToken accessToken, final OAut
...
-                    request.getAuthorities().stream().map(GrantedAuthority::getAuthority).collect(Collectors.toSet()));
+        final var username = getUsernameFromRequestParam(requestParams);
+        if (username.isPresent()) {
+            additionalInfo.put(ADD_INFO_USER_NAME, username.get());
+            additionalInfo.put(SUBJECT, username.get());
         } else {
-            additionalInfo.put(ADD_INFO_AUTH_SOURCE, "none");
             additionalInfo.put(SUBJECT, authentication.getName());
         }
+        additionalInfo.put(ADD_INFO_AUTH_SOURCE, getAuthSourceFromRequestParam(requestParams));
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/JWTTokenEnhancer.java:103]: allow auth source to be passed in, with default of `none`

File: src/test/groovy/uk/gov/justice/digital/hmpps/oauth2server/api/specs/OauthSpecification.groovy [hunk 3]
```diff
@@ -148,18 +158,6 @@ class OauthSpecification extends TestSpecification {
         token.additionalInformation.auth_source == 'none'
     }
 
-    def "Client Credentials Login failure for non auth or nomis user without scope priv"() {
```
Comment (petergphillips) [inline src/test/groovy/uk/gov/justice/digital/hmpps/oauth2server/api/specs/OauthSpecification.groovy:151]: not needed as don't validate username for client credentials any more

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/JWTTokenEnhancer.java [hunk 4]
```diff
@@ -95,29 +93,34 @@ public OAuth2AccessToken enhance(final OAuth2AccessToken accessToken, final OAut
...
-            final Set<String> scope = new HashSet<>(request.getScope());
-            scope.add("write");
-            ((DefaultOAuth2AccessToken) accessToken).setScope(scope);
-
-            // Merge user authorities with those associated with client credentials
-            additionalInfo.put(ADD_INFO_AUTHORITIES,
-                    request.getAuthorities().stream().map(GrantedAuthority::getAuthority).collect(Collectors.toSet()));
+        final var username = getUsernameFromRequestParam(requestParams);
+        if (username.isPresent()) {
+            additionalInfo.put(ADD_INFO_USER_NAME, username.get());
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/JWTTokenEnhancer.java:98]: was previously missing the `.get()` here

**Review by Mjwillis — APPROVED** [review]
LGTM

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**